README.md

```diff
@@ -186,9 +186,47 @@ IterableAPI.email = @"user@example.com";
 
 ### Sending push notifications
 
-See Apple's [UserNotifications framework](https://developer.apple.com/documentation/usernotifications) document for information about how to register for remote notifications.
+1. Request authorization and register for remote notifications
 
-1. Send a remote notification token to Iterable
+    iOS apps must request authorization to receive push notifications.
+    To do this, they should call the [`requestAuthorization`](https://developer.apple.com/documentation/usernotifications/unusernotificationcenter/1649527-requestauthorization)
+    method on [`UNNotificiationCenter`](https://developer.apple.com/documentation/usernotifications/unusernotificationcenter).
+
+    This will prompt the user for permission to receive push notifications 
+    (unless the app has requested permission for provisional notifications, 
+    which are delivered silently and don't require the user's permission).
+
+    For example:
+
+    *Swift*
+    
+    ```swift
+    UNUserNotificationCenter.current().requestAuthorization(options:[.alert, .badge, .sound]) { (success, error) in
+        // ...
+    }
+    ```
+    
+    *Objective-C*
+    
+    ```objc
+    UNUserNotificationCenter* center = [UNUserNotificationCenter currentNotificationCenter];
+    [center requestAuthorizationWithOptions: (UNAuthorizationOptionAlert + UNAuthorizationOptionBadge + UNAuthorizationOptionSound)
+       completionHandler:^(BOOL granted, NSError * _Nullable error) {
+       // ...
+    }];
+    ```
+    
+    By default, if you have set `IterableAPI.email` or `IterableAPI.userId`
+    (as described above), the SDK will then register the app for remote 
+    notifications automatically.
+    
```

Comment:
This is a little tricky. If you set userId or email. IterableSDK will register for token without asking for permission. If the app has background mode enabled then we can send silent pushes. I am not sure if we should make this clearer.

Reply:
This is a constant source of confusion.

Reply:
@tapashmajumder I updated this—could you take another look?